xarray/core/utils.py

```diff
@@ -96,6 +96,14 @@ def maybe_coerce_to_str(index, original_coords):
     return index
 
 
+def maybe_coerce_to_dict(dict_like):
```

Comment:
Am I missing something or does this always just do `dict(dict_like)` given the branch always evaluates to False?

Reply:
At the moment, yes. I just haven't figured out how the boolean check should look like yet.